setup.py

```diff
@@ -3,48 +3,86 @@
 from setuptools import setup, find_packages
 
 
-setup(
-    name='py-evm',
-    # *IMPORTANT*: Don't manually change the version here. Use the 'bumpversion' utility.
-    version='0.2.0-alpha.14',
-    description='Python implementation of the Ethereum Virtual Machine',
-    long_description_markdown_filename='README.md',
-    author='Piper Merriam',
-    author_email='piper@pipermerriam.com',
-    url='https://github.com/ethereum/py-evm',
-    include_package_data=True,
-    py_modules=['evm', 'trinity', 'p2p'],
-    install_requires=[
+deps={
+    'evm': [
         "aiohttp>=2.3.1,<3.0.0",
         "async_lru>=0.1.0,<1.0.0",
+        "coincurve>=7.0.0,<8.0.0",
```

Comment:
this should not be part of the default `py-evm` install.  It requires a system dependency and we are working to avoid as many system dependencies for the base library.

Reply:
Ah, so that explains the existing grouping that I couldn't make sense of :+1: 

However, I think I haven't fully grokked that yet.

> It requires a system dependency and we are working to avoid as many system dependencies for the base library

So, if I get you right you want that the `install_requires` (the ones taken into account for anyone running `pip install py-evm`) dependencies to not include `coincurve`, `eth-hash` and `plyvel` but still want these to be included for everyone running a `pip install -e .[dev]`. Is that correct?

Do you mind elaborating a bit on that? I understand that we want to avoid system dependencies for the base library but if **not installing** these dependencies means that the evm doesn't work how would that help?

Reply:
> but still want these to be included for everyone running a `pip install -e .[dev]`. Is that correct?

Yes.

The reasoning for having `pip install py-evm` install an absolute minimum number of requirements (and absolute minimal system requirements) is to make the base library **very** easy to install and use as a *library* with minimal assumptions about underlying things like what crypto implementations they want to use, what database backend they want to use, etc.

Reply:
>  if not installing these dependencies means that the evm doesn't work how would that help?

This is not the case.  `eth-keys` ships with a pure python backend for the ecc operations.  `coincurve` is just faster.  Same with `plyvel`.  py-evm doesn't care about what database backend it uses and it does just fine with a minimal `MemoryDB`.

Reply:
Ah `py-evm` has existing fallbacks for these libraries in case these are missing. That clears it up! :+1: